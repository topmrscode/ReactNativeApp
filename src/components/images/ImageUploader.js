import * as React from 'react';
// iNPORT POUR UPLOADER UNE PHOTO
import ImagePicker from 'react-native-image-picker';
import {
  Button,
  StatusBar,
  StyleSheet,
  View,
  Image, 
  Text
} from 'react-native';

import {PostImageOnApi} from '../../api/Image'
import BaseContainer from '../../assets/styles/Base';
import Icon from 'react-native-vector-icons/FontAwesome';


// ----------------------------------------------------------
// IMPORTER UNE IMAGE DEPUIS MON TELEPHONE : options => 
// - depuis la galleries d'images sur mon tel
// - prendre une photo
// ----------------------------------------------------------

class ImageUploader extends React.Component {
  constructor(props){
    super(props)

    this.state = {
      imageSource : ""
    }

    this.ShowPicker = this.ShowPicker.bind(this);

  }

  ShowPicker(){
    const options = {
      title: 'Select Image',
      storageOptions: {
        skipBackup: true,
        path: 'images',
      },
    };
    
    ImagePicker.showImagePicker(options, (response) => {
    
      if (response.didCancel) {
        console.log('User cancelled image picker');
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
      } else if (response.customButton) {
        console.log('User tapped custom button: ', response.customButton);
      } else {
        // appel de ma fonction POST pourr uploader l image esur IMGUR
        // une fois que je les mis sur imgur, je veux l afficher sur ma page
        PostImageOnApi(response).then((data) => {
          this.setState({imageSource : {uri : response.uri}})
        })
      }
    }); 
  }

    static navigationOptions = {
      title: 'Import an Image here',
    };
  
    render() {
      if (this.state.imageSource != "") {
      return (
        <BaseContainer title={'Picture Uploader'} font={''} fontSize={20}>
        <View style={styles.container}>
         <Icon  style={{color:"#3b5998"}} name={'upload'} size={100} onPress={this.ShowPicker} />
         <StatusBar barStyle="default" />
         <Image style={styles.image} source={this.state.imageSource} />
        </View>
        </BaseContainer>
      );
      }
      else {
        return (
          <BaseContainer title={'Picture Uploader'} font={''} fontSize={20}>
            <View style={styles.container}>
            <Icon  style={{color:"#3b5998"}} name={'upload'} size={100} onPress={this.ShowPicker} />
              <StatusBar barStyle="default" />
              <Text>Upload an image</Text>
            </View>
          </BaseContainer>
        );
      }
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor:"#dfe1ee"
    },
    image: {
      height : 400,
      width: 400
    }
});
    
export default ImageUploader;
