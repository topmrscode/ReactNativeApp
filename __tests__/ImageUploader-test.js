import React from 'react';
import ImageUploader from '../src/components/images/ImageUploader';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';


// ---------------------
//  RENDU DES DONNEES  
// ---------------------
// creer un snapshot de mon rendu pour mon component Feed pour m assurer
// que chaque fois que je lance mon rendu il match bien avec celui d'avant 

Enzyme.configure({ adapter: new Adapter()});
test('renders correctly', async () => {

  const wrapper = await (shallow(<ImageUploader  />));

  const tree = renderer.create(wrapper).toJSON();
  expect(tree).toMatchSnapshot();
  wrapper.instance().ShowPicker();
});