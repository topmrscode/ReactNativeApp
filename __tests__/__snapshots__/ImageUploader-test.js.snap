// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#323539",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {
          "elevation": 3,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "",
            "fontSize": 20,
          },
        ]
      }
    >
      Picture Uploader
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dfe1ee",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 100,
          },
          Object {
            "color": "#3b5998",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï‚“
    </Text>
    <Text>
      Upload an image
    </Text>
  </View>
</RCTSafeAreaView>
`;
